src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryService.scala

```diff
@@ -54,14 +57,51 @@ class LibraryService (protected val argUserInfo: UserInfo, val rawlsDAO: RawlsDA
   implicit val userInfo = argUserInfo
 
   override def receive = {
-    case UpdateAttributes(ns: String, name: String, attrsJsonString: String) => asCurator {updateAttributes(ns, name, attrsJsonString)} pipeTo sender
-    case SetPublishAttribute(ns: String, name: String, value: Boolean) => asCurator {setWorkspaceIsPublished(ns, name, value)} pipeTo sender
+    case UpdateAttributes(ns: String, name: String, attrsJsonString: String) => updateAttributes(ns, name, attrsJsonString) pipeTo sender
+    case UpdateDiscoverableBy(ns: String, name: String, newGroups: Seq[String]) => updateDiscoverableBy(ns, name, newGroups) pipeTo sender
+    case SetPublishAttribute(ns: String, name: String, value: Boolean) => setWorkspaceIsPublished(ns, name, value) pipeTo sender
     case IndexAll => asAdmin {indexAll} pipeTo sender
     case FindDocuments(criteria: LibrarySearchParams) => findDocuments(criteria) pipeTo sender
     case Suggest(criteria: LibrarySearchParams) => suggest(criteria) pipeTo sender
-    case PopulateSuggest(field: String, text: String) => asCurator {populateSuggest(field: String, text: String)} pipeTo sender
+    case PopulateSuggest(field: String, text: String) => populateSuggest(field: String, text: String) pipeTo sender
   }
 
+  def hasAccessOrCurator(workspaceResponse: WorkspaceResponse, neededLevel: WorkspaceAccessLevels.WorkspaceAccessLevel): Boolean = {
+    val wsaccess = workspaceResponse.accessLevel >= neededLevel
+    if (!wsaccess)
+      tryIsCurator(userInfo) map { value => return value }
+    wsaccess
+  }
+
```

Comment:
this function doesn't work because i can't return the value from inside the map (i.e. the future).
i would like some sort of convenience function because we do these types of checks in several places below... any suggestions on what to do?